kfs-prje/src/main/java/edu/arizona/kfs/module/prje/service/impl/PRJEServiceImpl.java

```diff
@@ -0,0 +1,857 @@
+            }
+        }
+        
+        private KualiDecimal processPRJETransferRecord(PRJETransferRecord transferRecord) {
+            KualiDecimal availableBalance = this.calculateBaseAccountAmount(transferRecord.getBaseAccount());
+            KualiDecimal amount = calculateTransfer(transferRecord, availableBalance);
+            
+            if ( amount.isNonZero() ) {
+                transferRecord.setBalance(availableBalance);
+                transferRecord.setAmount(amount);
+                
+                // Create the Origin Entries based on the Transfer Record Amount
+                OriginEntryFull debitEntry = generateDebitEntry(transferRecord);
+                OriginEntryFull creditEntry = generateCreditEntry(transferRecord);
+                
+                transferRecord.setDebitEntry(debitEntry);
+                transferRecord.setCreditEntry(creditEntry);
+                
+                processedTransfers.add(transferRecord);
+            }
+            else {
+                LOG.warn("--- Attempting to Transfer Zero");
+            }
+            
+            return amount;
+        }
+
+        private OriginEntryFull generateDebitEntry(PRJETransferRecord transferRecord) {
+            PRJEBaseAccount baseAccount = transferRecord.getBaseAccount();
+            KualiDecimal amount = transferRecord.getAmount();
+            KualiDecimal balance = transferRecord.getBalance();
+            
+            // From (Base) Side:
+            
+            // Debit the From (Base) Account for expense object code, 
+            // or credit the From (Base) Account for revenue object 
+            // code.  However, if the override chart and accounts are 
+            // defined use the amount under the From (Base) chart-account
+            // for calculation, but move the money from the override 
+            // chart-account.
+            
+            String account;
+            String subAccount;
+            String chartCode;
+
+            // Decide The Source of the Debit (From or Base)
+            if ( StringUtils.isNotBlank(baseAccount.getFromAccount()) ) {
+                account = baseAccount.getFromAccount();
+                subAccount = baseAccount.getFromSubAccount();
+                chartCode = baseAccount.getFromChart();
+            }
+            else {
+                account = baseAccount.getBaseAccount();
+                subAccount = baseAccount.getBaseSubAccount();
+                chartCode = baseAccount.getBaseChart();
+            }
+            
+            LOG.info("--- Debit " + amount + " from Base Account " + account);
+            
+            PRJEType type = baseAccount.getType();
+            
+            // Create the origin entry
+            OriginEntryFull debitEntry = new OriginEntryFull(PRJEConstants.ENTRY_DOCTYPE, PRJEConstants.ENTRY_ORIGIN);
+            debitEntry.setFinancialBalanceTypeCode(KFSConstants.BALANCE_TYPE_ACTUAL);
```

Comment:
KFSConstants.BALANCE_TYPE_ACTUAL is deprecated. You should get these values from the system options service. See edu.arizona.kfs.gl.batch.service.impl.BudgetAdjustmentCashTransferServiceImpl.